guides/common/modules/ref_pxe-loaders.adoc

```diff
@@ -0,0 +1,84 @@
+:_mod-docs-content-type: REFERENCE
+
+[id="pxe-loaders"]
+= PXE loaders
+
+In {Project}, a PXE loader defines the `filename` DHCPv4 option to use during provisioning.
+{Project} deploys the required files to TFTP {SmartProxy}.
+
+.Limitations
+* {Project} can deploy the required booting files only for the `x86_64` architecture.
+ifdef::satellite[]
+* {Team} does not provide support for iPXE workflows.
+endif::[]
+
+.PXE loaders and DHCPv4 filenames
+|===
+| PXE loader option | `filename` value | Use for
+
+| *None*
+| `""`
+| ?
+
+| *PXELinux BIOS*
+| `pxelinux.0`
+| BIOS machines with PXE booting
+
+| *PXELinux UEFI*
+| `pxelinux.efi`
+| UEFI machines with PXE booting
+
+| *Grub2 BIOS*
+| `grub2/grub__arch__.0`
+| BIOS machines with PXE booting
+
+| *Grub2 UEFI*
+| `__subdir__/grub2/grub__arch__.efi`
+| UEFI machines with PXE booting
+
+| *Grub2 UEFI SecureBoot*
+| `__subdir__/grub2/shim__arch__.efi`
+| SecureBoot-enabled UEFI machines with PXE booting
+
+| *Grub2 UEFI HTTP*
+| `http://__httpboot.example.com__/httpboot/__subdir__/grub2/grub__arch__.efi`
+| UEFI machines with HTTP booting
+
+| *Grub2 UEFI HTTPS*
+| `https://__httpboot.example.com__/httpboot/__subdir__/grub2/grub__arch__.efi`
+| UEFI machines with HTTP booting over HTTPS
+
+ifndef::satellite[]
+| *Grub2 UEFI HTTPS SecureBoot*
+| `https://__httpboot.example.com__/httpboot/__subdir__/grub2/shim__arch__.efi`
+| SecureBoot-enabled UEFI machines with HTTP booting over HTTPS
+endif::[]
+
+| *iPXE Embedded*
```

Comment:
I'll admin I don't know [iPXE](https://ipxe.org/) so this is mostly reverse engineering. I had hoped https://theforeman.org/manuals/3.15/index.html#4.4.3ProvisioningTemplates would document this but that is largely unchanged since 2018.

I can see https://github.com/theforeman/foreman/commit/c521c6e5bd19f2abbd30d51c0ad1a8e00f736fab implemented iPXE support.

Digging around I suspect this is for https://ipxe.org/embed and the way I read is that if you build your own iPXE image. You can embed the URL in there. The URL will be something like `http://foreman.example.com/unattended/iPXE`.

Reply:
@ShimShtein On a "when should I use this option"-level it might be somewhat clearer, but in a more holistic sense it's still somewhat lacking.  
I.e. "PXE Loader: iPXE Capable NIC(s)" doesn't read particularly well, and it might easily be understood as "Is my NIC capable of running iPXE?" - which is almost exclusively a "yes", instead of "Does my NIC use iPXE in its PXE/option ROM?" which is more likely to be a "no".

I don't personally have any really good suggestions either, only things that go more technical about it - "iPXE in PXE ROM" or maybe "iPXE in NIC Firmware"